desktopexporter/internal/store/queries.go

```diff
@@ -62,12 +62,25 @@ const (
 		statusCode VARCHAR, 
 		statusMessage VARCHAR)
 	`
-	SELECT_ORDERED_TRACES = `
-		SELECT traceID 
-		FROM spans
-		GROUP BY traceID
-		ORDER BY MAX(startTime) DESC
-	`
+	// SELECT_TRACE_SUMMARIES retrieves all traces, with root spans first.
+	// Uses COALESCE(startTime, far_future_date) in ORDER BY to handle NULL timestamps:
+	// - For traces with root spans: uses actual startTime
+	// - For traces without root spans: uses far future date (9999-12-31)
+	// This ensures traces without root spans appear after those with root spans in DESC order
```

Comment:
I'm not sure this is desired behavior. No root span likely means one of the following is true:

- the trace is still running and not finished yet
- something likely went wrong and the program crashed, especially during development. We likely don't want to hide that trace at the bottom.
- config error, the trace began elsewhere and isn't being forwarded to this service for some reason

Can we use the earliest timestamp in the trace maybe? 🤔

Reply:
You are totally right! I didn’t think about how actual usability here with genuinely broken traces here- I only thought about the unfinished ones and was like “Heh, it will pop back into order when the root gets there”. Good call.

 I just fixed it to sort by root span start if we have one, and fall back to easiest span start if we do not. Should be better.

Reply:
Also the robit helped me write tests! 😁